File: 1523-count-odd-numbers-in-an-interval-range/1523-count-odd-numbers-in-an-interval-range.py
class Solution:
    def countOdds(self, low: int, high: int) -> int:
        if low%2==1 and high%2==1:
            return (high-low)//2 + 1
        # both even
        elif low%2==0 and high%2==0:
            return (high-low)//2
        # one odd, one even
        else:
            return (high-low+1)//2

File: 1523-count-odd-numbers-in-an-interval-range/1523-count-odd-numbers-in-an-interval-range.java
class Solution {
    public int countOdds(int low, int high) {
        if(low==high)
        return low%2!=0?1:0;
        int c=0;
        if(low%2!=0)c++;
        if(high%2!=0)c++;
        int diff=(high-low)-1;
        c+=low%2!=0?diff/2:(diff%2!=0?(diff/2)+1:diff/2);
        return c;  
    }
}